### ISSUE TITLE
crash on libnice,nice_agent_peer_candidate_gathering_done

### ISSUE THREAD

**Issue by darrenhp** [Kurento/bugtracker#501]
## Prerequisites

* [x] I have read the SUPPORT document
  
```
root@kurento-online-bj-2020061005:/var/crash/1# apt list|grep kurento-dbg

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

kurento-dbg/bionic 6.14.0-0kurento1.18.04 amd64
```
* [x] I have checked the Troubleshooting Guide
  

* [] I have tested with the latest version of Kurento
  

## Issue description

crash, but i have no idea to reproduce it;
```
libnice:ERROR:conncheck.c:2272:priv_mark_pair_nominated: assertion failed (pair->state == NICE_CHECK_DISCOVERED): (0 == 6)
Aborted (thread 140559088367360, pid 1664)
Stack trace:
[gsignal]
/lib/x86_64-linux-gnu/libc.so.6:0x3EE97
[abort]
/lib/x86_64-linux-gnu/libc.so.6:0x40801
[g_assertion_message]
/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0:0x73315
[g_assertion_message_cmpnum]
/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0:0x736C2
[nice_agent_peer_candidate_gathering_done]
/usr/lib/x86_64-linux-gnu/libnice.so.10:0x19AE8
[nice_agent_peer_candidate_gathering_done]
/usr/lib/x86_64-linux-gnu/libnice.so.10:0x1C7B5
[nice_agent_gather_candidates]
/usr/lib/x86_64-linux-gnu/libnice.so.10:0x155BA
[nice_agent_gather_candidates]
/usr/lib/x86_64-linux-gnu/libnice.so.10:0x158D5
[g_socket_is_closed]
/usr/lib/x86_64-linux-gnu/libgio-2.0.so.0:0x7E0F9
[g_main_context_dispatch]
/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0:0x4C285
[g_main_context_dispatch]
/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0:0x4C650
[g_main_loop_run]
/usr/lib/x86_64-linux-gnu/libglib-2.0.so.0:0x4C962
[gst_nice_src_get_type]
/usr/lib/x86_64-linux-gnu/gstreamer-1.5/libgstnice15.so:0x2EA4
[gst_base_src_wait_playing]
/usr/lib/x86_64-linux-gnu/libgstbase-1.5.so.0:0x2EEA9
[gst_base_src_wait_playing]
/usr/lib/x86_64-linux-gnu/libgstbase-1.5.so.0:0x30B93
[gst_tag_setter_get_tag_merge_mode]
/usr/lib/x86_64-linux-gnu/libgstreamer-1.5.so.0:0xA0A10
```

## How to reproduce?

I don't know

### INFO about Kurento Media Server

* Kurento version:  6.14.0
* Server OS:        18.04 (Bionic)
* Installation method:
    
  - [x] apt-get
    
  - [] Docker
    
  - [] AWS
    
  - [] Built from sources
    

### INFO about your Application Server

* Language:                Java
* Kurento Client version:   

### INFO about end-user clients

* Device(s):    
* OS(es):       
* Browser(s):   

### Commands output

```
root@kurento-online-bj-2020061005:/var/crash/1# kurento-media-server --version
Kurento Media Server version: 6.14.0
Found modules:
	'core' version 6.14.0
	'elements' version 6.14.0
	'filters' version 6.14.0
root@kurento-online-bj-2020061005:/var/crash/1#
root@kurento-online-bj-2020061005:/var/crash/1# dpkg -l | grep -Pi 'kurento|kms-|gst.*1.5|nice'
ii  gstreamer1.5-libav:amd64               1.8.1-1kurento1.18.04                           amd64        libav plugin for GStreamer
ii  gstreamer1.5-nice:amd64                0.1.17-0kurento1.18.04                          amd64        ICE library (GStreamer 1.5 plugin)
ii  gstreamer1.5-plugins-bad:amd64         1.8.1-1kurento4.18.04                           amd64        GStreamer plugins from the "bad" set
ii  gstreamer1.5-plugins-base:amd64        1.8.1-1kurento2.18.04                           amd64        GStreamer plugins from the "base" set
ii  gstreamer1.5-plugins-good:amd64        1.8.1-1kurento4.18.04                           amd64        GStreamer plugins from the "good" set
ii  gstreamer1.5-plugins-ugly:amd64        1.8.1-1kurento1.18.04                           amd64        GStreamer plugins from the "ugly" set
ii  gstreamer1.5-pulseaudio:amd64          1.8.1-1kurento4.18.04                           amd64        GStreamer plugin for PulseAudio
ii  kms-core                               6.14.0-0kurento1.18.04                          amd64        Kurento Core module
ii  kms-elements                           6.14.0-0kurento1.18.04                          amd64        Kurento Elements module
ii  kms-filters                            6.14.0-0kurento1.18.04                          amd64        Kurento Filters module
ii  kms-jsonrpc                            6.14.0-0kurento1.18.04                          amd64        Kurento JSON-RPC library
ii  kmsjsoncpp                             1.6.3-1kurento1.18.04                           amd64        Kurento jsoncpp library
ii  kurento-media-server                   6.14.0-0kurento1.18.04                          amd64        Kurento Media Server
ii  libgstreamer-plugins-bad1.5-0:amd64    1.8.1-1kurento4.18.04                           amd64        GStreamer development files for libraries from the "bad" set
ii  libgstreamer-plugins-base1.5-0:amd64   1.8.1-1kurento2.18.04                           amd64        GStreamer libraries from the "base" set
ii  libgstreamer1.5-0:amd64                1.8.1-1kurento2.18.04                           amd64        Core GStreamer libraries and elements
ii  libnice10:amd64                        0.1.17-0kurento1.18.04                          amd64        ICE library (shared library)
ii  libsrtp0:amd64                         1.6.0-0kurento1.18.04                           amd64        Secure RTP (SRTP) and UST Reference Implementations - shared library
ii  libusrsctp                             0.9.2-1kurento1.18.04                           amd64        sctp-refimpl library
ii  openh264                               1.4.0-1kurento1.18.04                           amd64        OpenH264 library
ii  openh264-gst-plugins-bad-1.5:amd64     1.8.1-1kurento4.18.04                           amd64        GStreamer plugins from openh264
ii  openwebrtc-gst-plugins                 0.10.0-1kurento1.18.04                          amd64        OpenWebRTC specific GStreamer plugins
root@kurento-online-bj-2020061005:/var/crash/1#

```

**Comment by j1elo** [comment]
Please have the debug symbols installed so the crash produces a stack trace with file names and line numbers. I can see they are not installed as otherwise we'd see "kms-core-dbg", "kurento-media-server-dbg" etc. packages in the dpkg command output.

If you have or find out about any way about how to reproduce it, or under what circumstances it happens, it will be easier to study and fix. E.g. with high load, or low load, long time since startup, clients connecting from Chrome, Firefox or Safari, etc.

**Comment by darrenhp** [comment]
Actually, we have installed the debug by `sudo apt-get update && sudo apt-get install --no-install-recommends --yes \
    kurento-dbg`
```
root@kurento-online-bj-2020061005:/var/crash# apt list|grep 'kurento\|kms'|grep dbg

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

gstreamer1.0-nice-dbgsym/bionic 0.1.17-0kurento1.18.04 amd64
gstreamer1.5-libav-dbg/bionic 1.8.1-1kurento1.18.04 amd64
gstreamer1.5-nice-dbgsym/bionic 0.1.17-0kurento1.18.04 amd64
gstreamer1.5-plugins-bad-dbg/bionic 1.8.1-1kurento4.18.04 amd64
gstreamer1.5-plugins-base-dbg/bionic 1.8.1-1kurento2.18.04 amd64
gstreamer1.5-plugins-good-dbg/bionic 1.8.1-1kurento4.18.04 amd64
gstreamer1.5-plugins-ugly-dbg/bionic 1.8.1-1kurento1.18.04 amd64
kms-core-dbg/bionic 6.14.0-0kurento1.18.04 amd64
kms-datachannelexample-dbg/bionic 6.7.0.18.04 amd64
kms-elements-dbg/bionic 6.14.0-0kurento1.18.04 amd64
kms-filters-dbg/bionic 6.14.0-0kurento1.18.04 amd64
kms-jsonrpc-dbg/bionic 6.14.0-0kurento1.18.04 amd64
kms-platedetector-dbg/bionic 6.7.0.18.04 amd64
kmsjsoncpp-dbg/bionic 1.6.3-1kurento1.18.04 amd64
kurento-dbg/bionic 6.14.0-0kurento1.18.04 amd64
kurento-media-server-dbg/bionic 6.14.0-0kurento1.18.04 amd64
libgstreamer1.5-0-dbg/bionic 1.8.1-1kurento2.18.04 amd64
libnice10-dbgsym/bionic 0.1.17-0kurento1.18.04 amd64
libsrtp0-dbg/bionic 1.6.0-0kurento1.18.04 amd64
libusrsctp-dbgsym/bionic 0.9.2-1kurento1.18.04 amd64
openwebrtc-gst-plugins-dbg/bionic 0.10.0-1kurento1.18.04 amd64
```
But the errors.log don't show stack like  `https://doc-kurento.readthedocs.io/en/6.14.0/dev/dev_guide.html#install-debug-symbols`
I happen

**Comment by darrenhp** [comment]
I try to reinstall the debug symbols; and found some warnings;

**kurento-dbg : Depends: libglib2.0-0-dbgsym but it is not installable or
                        libglib2.0-0-dbg but it is not installable**
```
root@kurento-online-bj-20200525003:~# sudo apt-get install --no-install-recommends --yes     kurento-dbg
Reading package lists... Done
Building dependency tree
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
kurento-dbg : Depends: libglib2.0-0-dbgsym but it is not installable or
                        libglib2.0-0-dbg but it is not installable
E: Unable to correct problems, you have held broken packages.
```

**Comment by j1elo** [comment]
Let me test in a clean container to make sure it works

**Comment by j1elo** [comment]
Hi @darrenhp I tested the installation of packages, to check if we have some problem in our repository. That's something that has happened in the past because of some mistake in the CI job scripts. But it is not the case this time; package installation works fine.

I tested with this all-in-one Docker command:

```
# In-place Docker container commands BEGIN
docker run --rm -i \
    "ubuntu:18.04" /bin/bash <<'DOCKERCOMMANDS'

# Bash options for strict error checking
set -o errexit -o errtrace -o pipefail -o nounset

# Trace all commands
set -o xtrace

# Install required tools
DEBIAN_FRONTEND=noninteractive \
apt-get update && apt-get install --no-install-recommends --yes \
    gnupg

# Add Kurento repository
apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 5AFA7A83
tee "/etc/apt/sources.list.d/kurento.list" >/dev/null <<EOF
deb [arch=amd64] http://ubuntu.openvidu.io/6.14.0 bionic kms6
EOF

# Install KMS
apt-get update && apt-get install --no-install-recommends --yes \
    kurento-media-server

# Install debug symbols
apt-key adv \
    --keyserver keyserver.ubuntu.com \
    --recv-keys F2EDC64DC5AEE1F6B9C621F0C8CAB6595FDFF622
source /etc/upstream-release/lsb-release 2>/dev/null || source /etc/lsb-release
tee "/etc/apt/sources.list.d/ddebs.list" >/dev/null <<EOF
deb http://ddebs.ubuntu.com ${DISTRIB_CODENAME} main restricted universe multiverse
deb http://ddebs.ubuntu.com ${DISTRIB_CODENAME}-updates main restricted universe multiverse
EOF
apt-get update && apt-get install --no-install-recommends --yes \
    kurento-dbg

echo "Everything got installed successfully!"

echo "KMS version:"
kurento-media-server --version

echo "List of DEBUG packages:"
dpkg -l | grep -Pi 'kurento|kms-|gst.*1.5|nice' | grep dbg

DOCKERCOMMANDS
# In-place Docker container commands END
```

The output of `echo` commands was:

```
Everything got installed successfully!

KMS version:
Kurento Media Server version: 6.14.0
Found modules:
	'core' version 6.14.0
	'elements' version 6.14.0
	'filters' version 6.14.0

List of DEBUG packages:
ii  gstreamer1.5-libav-dbg:amd64         1.8.1-1kurento1.18.04               amd64        libav plugin for GStreamer (debug symbols)
ii  gstreamer1.5-nice-dbgsym:amd64       0.1.17-0kurento1.18.04              amd64        debug symbols for gstreamer1.5-nice
ii  gstreamer1.5-plugins-bad-dbg:amd64   1.8.1-1kurento4.18.04               amd64        GStreamer plugins from the "bad" set (debug symbols)
ii  gstreamer1.5-plugins-base-dbg:amd64  1.8.1-1kurento2.18.04               amd64        GStreamer plugins from the "base" set
ii  gstreamer1.5-plugins-good-dbg:amd64  1.8.1-1kurento4.18.04               amd64        GStreamer plugins from the "good" set
ii  gstreamer1.5-plugins-ugly-dbg:amd64  1.8.1-1kurento1.18.04               amd64        GStreamer plugins from the "ugly" set (debug symbols)
ii  kms-core-dbg                         6.14.0-0kurento1.18.04              amd64        Kurento Core module
ii  kms-elements-dbg                     6.14.0-0kurento1.18.04              amd64        Kurento Elements module
ii  kms-filters-dbg                      6.14.0-0kurento1.18.04              amd64        Kurento Filters module
ii  kms-jsonrpc-dbg                      6.14.0-0kurento1.18.04              amd64        Kurento JSON-RPC library
ii  kmsjsoncpp-dbg                       1.6.3-1kurento1.18.04               amd64        Kurento jsoncpp library
ii  kurento-dbg                          6.14.0-0kurento1.18.04              amd64        Meta-package that installs debug symbols
ii  kurento-media-server-dbg             6.14.0-0kurento1.18.04              amd64        Kurento Media Server
ii  libgstreamer1.5-0-dbg:amd64          1.8.1-1kurento2.18.04               amd64        Core GStreamer libraries and elements
ii  libnice10-dbgsym:amd64               0.1.17-0kurento1.18.04              amd64        debug symbols for libnice10
ii  openwebrtc-gst-plugins-dbg           0.10.0-1kurento1.18.04              amd64        OpenWebRTC specific GStreamer plugins
```

So as you can see, the installation works perfectly and all Kurento debug packages are there from the dpkg command.

I guess at some point you broke your installation. I'd recommend to reinstall the machine (if it is a virtual environment) or start from a clean container.

If you borked your desktop system, I can suggest to follow the **Option B** from [Local Upgrade](https://doc-kurento.readthedocs.io/en/latest/user/installation.html#local-upgrade) section.

A *last recourse* is to run the command described in [Clean up your system](https://doc-kurento.readthedocs.io/en/latest/dev/dev_guide.html#clean-up-your-system) **BUT** that's **on your own risk**, if you run that command you should first review that it won't try to uninstall your whole system.

**Comment by darrenhp** [comment]
access ubuntu.openvidu.io from china is too slow；

**Comment by j1elo** [comment]
Related libnice issues:
- https://gitlab.freedesktop.org/libnice/libnice/-/issues/94
- https://gitlab.freedesktop.org/libnice/libnice/-/issues/95

libnice issue 95 was closed via commit [6afcb580](https://gitlab.freedesktop.org/libnice/libnice/-/commit/6afcb580ae8118e189b24c74f0832c8cfff39f4e), which was backported to Kurento in https://github.com/Kurento/libnice/commit/6afcb580ae8118e189b24c74f0832c8cfff39f4e
